extensions/pyRevitTools.extension/checks/radar_check.py

```diff
@@ -0,0 +1,624 @@
+    output.add_style('cover {color:black; font-size:24pt; font-weight:bold;}')
+    output.add_style('header {color:black; font-size:15pt;}')
+    output.print_html(
+    ('<cover>__________:satellite_antenna:__10-Mile Radar___________</cover>'))
+    print(divider)
+    print("")
+    output.print_md('# Checking model placement and coordinates')
+    print(divider)
+    # _____________________________Check the distances of base and survey points
+    baspt, survpt = get_project_base_and_survey_pts(document)
+    basptdistance = abs(calculate_distance(baspt, INTERNAL_ORIGIN))
+    surveydistance = abs(calculate_distance(survpt, INTERNAL_ORIGIN))
+
+    if basptdistance > EXTENT_DISTANCE:
+        output.print_md('{} Base Point is more than {}'.format(BAD_STRG, CRITERIA_STRG))
+    if surveydistance > EXTENT_DISTANCE:
+        output.print_md('{} Survey Point is more than {}'.format(BAD_STRG, CRITERIA_STRG))
+    else:
+        output.print_md('{} Survey Point is less than {}'.format(GOOD_STRG, CRITERIA_STRG))
+    basptdistance = calculate_distance(baspt, INTERNAL_ORIGIN)
+    # ____________________________________Calculate the angle between the points
+    baseptangle = calculate_angle(baspt, INTERNAL_ORIGIN)
+    surveyptangle = calculate_angle(survpt, INTERNAL_ORIGIN)
+    truenorthangle = calculate_angle(baspt, survpt)
+    # _______________________________Print the Project Coordinates and Distances
+    tbdata = [['Internal Origin Coordinates', 
+        str(INTERNAL_ORIGIN[0]),str(INTERNAL_ORIGIN[1]),str(INTERNAL_ORIGIN[2]),
+        ' ', ' '],
+        ['Project Base Point Coordinates to Internal Origin', 
+        str(convert_values(baspt[0],doc)),
+        str(convert_values(baspt[1],doc)),
+        str(convert_values(baspt[2], doc)), 
+        str(convert_units(basptdistance, document)), baseptangle],
+        ['Survey Point Coordinates to Internal Origin', 
+        str(convert_values(survpt[0], doc)),
+        str(convert_values(survpt[1], doc)),
+        str(convert_values(survpt[2], doc)),
+        str(convert_units(surveydistance, document)), surveyptangle],
+        ['Project Base Point to Survey Delta X', ' ', ' ', ' ',
+        str(convert_units(calculate_delta_x(
+            baspt, survpt), document))],
+        ['Project Base Point to Survey Delta Y', ' ', ' ', ' ',
+        str(convert_units(calculate_delta_y(
+            baspt, survpt), document))],
+        ['Planar Distance between Base Point and Survey Point',
+        ' ', ' ', ' ', 
+        str(convert_units(calculate_horizontal_distance(
+            baspt, survpt), document)),
+            truenorthangle],
+        ['Total Distance between Base Point and Survey Point',
+        ' ', ' ', ' ',
+        str(convert_units(calculate_distance(
+            baspt, survpt), document)),
+            ' '],
+        ['Project Elevation', 
+        ' ', ' ', str(convert_values(baspt[2], doc)),
+        str(convert_units((survpt[2] - baspt[2]), document))]]
```

Comment:
This is a formatting nightmare, and could be simplified.
I personally prefer the black/ruff style formatting (if a list is too long, brackets are on their own lines and the should be one per line, indented from the brackets), but I understand that it is not the general code style inside pyrevit (because we never had one :sweat_smile: )

since the `convert_unit` and `convert_value` are always inside a `str`, why not move the string conversion inside those functions and drop the `str` here?
The second and third line are the same, it is just the point and the distance that differs; they can be turned into a function that returns the line (the angle can be calculated inside that very function)
There's still the `doc` / `document`  variables problem.

Reply:
does this function below makes sense? I am getting multiple returns:

```
def points_distances_angles(point=pt):
    """
    Calculates the distances and angles between the internal origin and a point.
    Args:
        point (tuple): The coordinates of the point.
    Returns:
        distances (tuple): The distances between the internal origin and the point.
        angles (tuple): The angles between the internal origin and the point.
    """
    distances = calculate_distance(point, INTERNAL_ORIGIN)
    angles = calculate_angle(point, INTERNAL_ORIGIN)
    return distances, angles
```

Reply:
The point I was making for multiple returns is not that it is forbidden, but it is useless to have multiple returns if you then use only one of the returned values, as it was done in the original script. Multiple returns are OK if they are used together by the calling function.
What I meant in this particular case, though, is this

```python
def report_point(point, distance, document):
    angle = calculate_angle(point, INTERNAL_ORIGIN)
    return (convert_values(point[0], document), convert_values(point[1], document), convert_values(point[2], document), convert_units(distance, document), angle)
```

so you then can use

```python
tbdata = [
    # ...
    ['Project Base Point Coordinates to Internal Origin', *report_point(baspt, basptdistance, document)],
    ['Survey Point Coordinates to Internal Origin', *report_point(survpt, surveydistance, document)],
    # ...
]
```

Another thing that can be done is to transform the `convert_values` to a function that accepts any number of values to convert and convert all the coordinates at once:

```python
def convert_values(*values, document=doc):
    # ui_units logic unchanged...
    # ...
    return [
        "{:.3g}".format(DB.UnitUtils.ConvertFromInternalUnits(value, ui_units))
        for value in values
    ]
```

and then

```python
def report_point(point, distance, document):
    angle = calculate_angle(point, INTERNAL_ORIGIN)
    return (*convert_values(*point, document=document), convert_units(distance, document), angle)
```

you could even parametrize the message and add it to the `report_point` function, so that you can just use

```python
tbdata = [
    # ...
    report_point("Project Base Point ", baspt, basptdistance, document),
    report_point("Survey Point", survpt, surveydistance, document),
    # ...
]
```

Reply:
this is crazy good!!
I have this table function to cleanup, I learned a lot of good tricks already.

Reply:
this unpacking technique is very efficient, I should learn it more and capitalize on it. thank you for sharing.